File: interfaces.java
interface Vehiculo{
    void arrancar();
}

class Carro implements Vehiculo{
    public void arrancar(){
        System.out.println("El carro arranca");
    }
}

class Moto implements Vehiculo{
    public void arrancar(){
        System.out.println("La moto arranca");
    }
}

public class Main{
    public static void main(String[] args){
        Vehiculo v = new Carro();
        v.arrancar();
    }
}

File: interfaces.py
from abc import ABC, abstractmethod

class Vehiculo(ABC):

    @abstractmethod
    def arrancar(self):
        pass

class Carro(Vehiculo):
    def arrancar(self):
        print("El carro arranca")

class Moto(Vehiculo):
    def arrancar(self):
        print("La moto arranca")

v1 = Carro()
v1.arrancar()
